network/roundrobin.go

```diff
@@ -24,6 +29,32 @@ func (r *RoundRobin) NextProxy(_ IConnWrapper) (IProxy, *gerr.GatewayDError) {
 		return nil, gerr.ErrNoProxiesAvailable
 	}
 
-	nextIndex := r.next.Add(1)
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	// Get current index from Raft FSM
+	currentIndex := r.server.RaftNode.Fsm.GetRoundRobinNext(r.server.GroupName)
+	nextIndex := currentIndex + 1
+
+	// Create Raft command
+	cmd := raft.Command{
+		Type: raft.CommandAddRoundRobinNext,
+		Payload: raft.RoundRobinPayload{
+			NextIndex: nextIndex,
+			GroupName: r.server.GroupName,
+		},
+	}
+
+	// Convert command to JSON
+	data, err := json.Marshal(cmd)
+	if err != nil {
+		return nil, gerr.ErrNoProxiesAvailable.Wrap(err)
+	}
+
+	// Apply through Raft
+	if err := r.server.RaftNode.Apply(data, raft.LeaderElectionTimeout); err != nil {
```

Comment:
Should we use another timeout config parameter for this one?

Reply:
Addressed in [8fbdb8a](https://github.com/gatewayd-io/gatewayd/pull/641/commits/8fbdb8aca001d6cd24a3e1759699fe3dcca9b242)